### PR TITLE
[IMP] mail: send channel update via bus notification

### PR DESCRIPTION
**Pull request by phenix-factory**
Before this PR, channel update were not sent via bus notifications.

Part of task-2821415

### CODE REVIEW

File: addons/mail/static/tests/discuss_app/sidebar_tests.js [hunk 1]
```diff
@@ -1158,3 +1158,19 @@ QUnit.test("Unpinning channel closes its chat window", async () => {
...
+    await contains(".o-mail-DiscussSidebarChannel", { text: "Sales" });
+    pyEnv["bus.bus"]._sendone(pyEnv.currentPartner, "mail.record/insert", {
+        Thread: {
+            id: channelId,
+            model: "discuss.channel",
+            name: "test",
+        },
+    });
+    await contains(".o-mail-DiscussSidebarChannel", { text: "test" });
+});
```
Comment (alexkuhn) [inline addons/mail/static/tests/discuss_app/sidebar_tests.js:1176]: You could make a multi-tab test, and have one tab change the name, and see the name is changed on the other tab.
The test would need to adapt mock server accordingly, as you changed the code in python this makes sense

File: addons/mail/models/discuss/discuss_channel.py [hunk 2]
```diff
@@ -273,19 +273,32 @@ def write(self, vals):
...
+            for key in vals.keys():
+                if current_val.get(key) != vals.get(key) and key != "image_128":
+                    diff[key] = vals[key]
+                    notifications.append([channel, "mail.record/insert", {
+                        "Thread": {
+                            "id": current_val["id"],
+                            "model": "discuss.channel",
+                            **diff
+                        }
+                    }])
```
Comment (alexkuhn) [inline addons/mail/models/discuss/discuss_channel.py:289]: Slightly out-of-scope of PR, but I think `image_128` could be grouped in same notification

**Review by alexkuhn — COMMENTED** [review]
LGTM otherwise!

File: addons/mail/models/discuss/discuss_channel.py (hunk 2 above)
Comment (phenix-factory) [reply addons/mail/models/discuss/discuss_channel.py:289]: Yes and no, as I said, you don't want the image_128 field in the bus notification as base64. You want to fill the `avatarCacheKey`.
You could try to group the notification, but I m not sure the code would be better. Having a dedicated logic for `image_128` isn't that bad.

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

**Comment by alexkuhn** [comment]
@robodoo override=ci/security
sendmany on discuss.channel so ok

**Comment by alexkuhn** [comment]
@robodoo r+

**Pull request closed without merging**